pkg/test/ginkgo/external.go

```diff
@@ -0,0 +1,131 @@
+package ginkgo
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	imagev1 "github.com/openshift/api/image/v1"
+	"github.com/openshift/origin/test/extended/util"
+)
+
+type serializedTest struct {
+	Name   string
+	Labels string
+}
+
+// externalTestsForSuite reads tests from external binary, currently only
+// k8s-tests is supported
+func externalTestsForSuite(ctx context.Context) ([]*testCase, error) {
+	var tests []*testCase
+
+	// TODO: add support for binaries from other images
+	testBinary, err := extractBinaryFromReleaseImage("hyperkube", "/usr/bin/k8s-tests")
```

Comment:
Just a question to learn; as far as I can see hyperkube image was removed from k/k and we are using it in o/k?

Reply:
Yes we are, see https://github.com/openshift/kubernetes/tree/master/openshift-hack/images/hyperkube It's a simple script wrapper, since we don't plan to ship different k8s binaries separately. Not yet, at least :sweat_smile:

Reply:
I have the same question ^ too

Reply:
I didn't plan users to need that ever, for those local-dev/custom cases I'd probably just modify the vendored tests, built the openshift-tests binary and run it with the envvar to use the embedded tests. That seems the simplest, but I'm open to suggestions.